test/unit/creation_test.rb

```diff
@@ -303,7 +307,7 @@ def test_create_with_relationships_in_payload
             }
         }.to_json)
 
-    article = Article.new(title: 'Rails is Omakase', relationships: {comments: [Comment.new(id: 2)]})
+    article = Article.new(title: 'Rails is Omakase', relationships: {comments: [Comment.new(id: '2')]})
```

Comment:
it's not significant behavior change because this test never actually work without bug that I'm trying to fix.
current behavior it that if you add `id` as `integer` to relationship it will be added to payload as is.

I think we should convert all ids into strings because JSONAPI spec says that id *MUST* be a string
but it's an idea of different PR. this PR fixes another problem

Reply:
Should we cast `.to_s` as to avoid interface changes? Won't this break everyone who is upgrading from an older version?